src/main/java/io/github/mivek/model/WeatherCondition.java

```diff
@@ -89,7 +89,7 @@ public void addPhenomenon(final Phenomenon pPhenomenon) {
      * @return true if there is at least phenomenon.
      */
     public boolean isValid() {
-        return !phenomenons.isEmpty() || (intensity != null && descriptive != null);
+        return !phenomenons.isEmpty() || descriptive != null;

```

Comment:
I think that is still too permissive. According to the `Federal meteorological handbook` only the descriptive `SH` is valid as a WeatherCondition.